microprofile/microprofile-fault-tolerance-2.1/src/main/java/org/wildfly/swarm/ts/microprofile/fault/tolerance/v21/retry/RetryService.java

```diff
@@ -0,0 +1,165 @@
+package org.wildfly.swarm.ts.microprofile.fault.tolerance.v21.retry;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.enterprise.context.RequestScoped;
+
+import org.eclipse.microprofile.faulttolerance.Asynchronous;
+import org.eclipse.microprofile.faulttolerance.Fallback;
+import org.eclipse.microprofile.faulttolerance.Retry;
+import org.wildfly.swarm.ts.microprofile.fault.tolerance.v21.MyConnection;
+import org.wildfly.swarm.ts.microprofile.fault.tolerance.v21.MyException;
+
+@Asynchronous // all is async with fallback method
+@RequestScoped
+public class RetryService {
+
+    private AtomicInteger counter = new AtomicInteger(1);
+
+    // First fail, second pass - there will be 1 retry before success
+    @Retry(retryOn = Throwable.class)
+    @Fallback(fallbackMethod = "processFallback")
+    public CompletionStage<MyConnection> throwable_retryOnT() throws Throwable {
+        final Integer invocationNumber = counter.getAndIncrement();
+        if (invocationNumber % 2 == 1) {
+            throw new Throwable("Simulated Throwable");
+        }
+
+        return CompletableFuture.completedFuture(new MyConnection() {
+            @Override
+            public String getData() {
+                return "Hello from throwable_retryOnT, counter: " + invocationNumber;
+            }
+        });
+    }
+
+    // First fail, second pass - there will be 1 retry before success (Throwable is parent of Exception)
+    @Retry(retryOn = Throwable.class)
+    @Fallback(fallbackMethod = "processFallback")
+    public CompletionStage<MyConnection> myException_retryOnT() throws MyException {
```

Comment:
Is the difference between `throwable_retryOnT` and `myException_retryOnT` only the `throws` clause of the method declaration? Because that is totally ignored by Fault Tolerance.

Reply:
Ah, I see it now -- the difference is in which exception is thrown. OK, cool.